core_brain/trader_loop.py

```
@@ -568,6 +612,28 @@ def _cancel_orders(client, registry, orders) -> int:
 
+def _venue_resting_order_ids(client) -> Optional[set[str]]:
+    """Venue order ids currently resting, or None when the venue cannot say.
+
+    Returning None rather than an empty set on failure is the whole point: an
+    unreachable venue must not read as "nothing is resting", which would let a
+    replacement go out on top of a live order. Mirrors `venue.open_notional`,
+    which refuses the same way for the MAX_TOTAL_USD cap.
+    """
+    try:
+        orders = client.get_open_orders() or []
+    except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)